src/main.rs

```diff
@@ -125,93 +137,217 @@ const FIXED_UI_KEY: &[&str] = &[
+
+    let (ui_repo, fr_repo) = prepare_repositories(ui_local, fr_local, &repo_manager)?;
+    checkout_frontary(
+        args.frontary_path.as_ref(),
+        ui_repo.as_path(),
+        fr_repo.as_path(),
+    )?;
+    process_keys(&ui_repo, &fr_repo)?;
+    Ok(())
+}
 
-    if let Err(e) = setup_ssh_agent(&ssh_key_path) {
+fn initialize_ssh(ssh_key: &Path) -> Result<(), io::Error> {
+    if let Err(e) = setup_ssh_agent(ssh_key) {
         eprintln!("❌ Failed to set up SSH agent: {e}");
         return Err(io::Error::other("SSH setup failed"));
     }
+    println!("✅ SSH authentication succeeded");
+    Ok(())
+}
 
-    println!("🛠️ Cloning repository: {frontary_url}...");
-    if let Err(e) = repo_manager.clone_repo(frontary_url, "frontary") {
-        eprintln!("Failed to clone frontary: {e}");
-        return Err(io::Error::other(format!(
-            "❌ Failed to clone frontary: {e}"
-        )));
-    }
-    println!("🛠️ Cloning repository: {ui_url}...");
-    if let Err(e) = repo_manager.clone_repo(ui_url, "aice-web") {
-        eprintln!("Failed to clone aice-web: {e}");
-        return Err(io::Error::other(format!(
-            "❌ Failed to clone aice-web: {e}"
-        )));
+fn log_repo_strategy(ui_path: Option<&PathBuf>, fr_path: Option<&PathBuf>) {
+    match (ui_path.as_ref(), fr_path.as_ref()) {
+        (None, None) => {
+            println!("🔄 No local paths: will clone both 'aice-web' and 'frontary'.");
+        }
+        (Some(path), None) => {
+            println!(
+                "🔄 Using local aice-web at {}; will clone frontary.",
+                path.display()
+            );
+        }
+        (None, Some(path)) => {
+            println!(
+                "🔄 Will clone aice-web; using local frontary at {}.",
+                path.display()
+            );
+        }
+        (Some(ui_path), Some(fr_path)) => {
+            println!(
+                "🔄 Using local aice-web at {} and frontary at {}.",
+                ui_path.display(),
+                fr_path.display()
+            );
+        }
     }
+}
 
-    let temp_path = repo_manager.temp_dir.path();
-    // Update file paths dynamically using `temp_path`
-    let en_path = temp_path.join("aice-web/langs/en-US.json");
-    let ko_path = temp_path.join("aice-web/langs/ko-KR.json");
-    let ui_files = get_files_with_extension(temp_path.join("aice-web/src"), "rs")?;
-    let css_files = get_files_with_extension(temp_path.join("aice-web/static"), "css")?;
-    let frontary_file_paths = get_files_with_extension(temp_path.join("frontary"), "rs")?;
-    let css_classes_and_ids = extract_css_classes_and_ids(&css_files)?;
+fn prepare_repositories(
+    ui_local: Option<PathBuf>,
+    fr_local: Option<PathBuf>,
+    mgr: &RepoManager,
+) -> Result<(PathBuf, PathBuf), io::Error> {
+    // aice-web
+    let ui_repo = if let Some(path) = ui_local {
+        println!("📂 Using local aice-web at {}", path.display());
```

Comment:
1.How about simplifying this part by removing the extra logs, since log_repo_strategy already provides sufficient details?

```rust
let ui_repo = prepare_repo(
    "git@github.com:aicers/aice-web.git",
    ui_local,
    "aice-web",
    mgr,
)?;
```

Reply:
Since the code clearly indicates which repository is being handled, the comment stating the repository name seems unnecessary. Would you mind removing it?